internal/backends/postgresql/collection.go

```diff
@@ -251,8 +251,42 @@ func (c *collection) Explain(ctx context.Context, params *backends.ExplainParams
 
 // Stats implements backends.Collection interface.
 func (c *collection) Stats(ctx context.Context, params *backends.CollectionStatsParams) (*backends.CollectionStatsResult, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/3398
-	return new(backends.CollectionStatsResult), nil
+	p, err := c.r.DatabaseGetExisting(ctx, c.dbName)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if p == nil {
+		return nil, backends.NewError(
+			backends.ErrorCodeDatabaseDoesNotExist,
```

Comment:
In a similar situation for `ListIndexes` we decided to only return `ErrorCodeCollectionDoesNotExist` because these errors - `DatabaseDoesNotExist` and `CollectionDoesNotExist` had the same usage in the handlers. I checked the usage for `Stats`' errors, and it's also the same: in the handlers, the behavior for both codes is the same, so we don't need to have two different codes. With that, I suggest doing in `Stats` the same thing we did in `ListIndexes` (see `func (cc *collectionContract) ListIndexes`).